flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  description = "Future-proof note-taking and publishing based on Zettelkasten";
+
+  inputs = {
+    # Please sync revision with ./dep/nixpkgs/github.json
+    nixpkgs.url =
+      "github:nixos/nixpkgs/fcab19deb78fbb5ea24e19b133cf34358176396a";
```

Comment:
Can `edolstra/flake-compat` be used to avoid having to do this? See https://nixos.wiki/wiki/Flakes#Using_flakes_project_from_a_legacy_Nix

I'd ensure DRY when switching to flakes.

More generally, we can get rid of `./dep/nixpkgs` allowing flakes to pin nixpkgs if that's doable.

Reply:
Sure! That sound good. Let me try the `flake-compat` solution first.